// @mempool/mempool.js/lib/index.js
"use strict";
var index_1 = require("./services/api/index");
var addresses_1 = require("./app/bitcoin/addresses");
var blocks_1 = require("./app/bitcoin/blocks");
var difficulty_1 = require("./app/bitcoin/difficulty");
var fees_1 = require("./app/bitcoin/fees");
var lightning_1 = require("./app/bitcoin/lightning");
var mempool_1 = require("./app/bitcoin/mempool");
var transactions_1 = require("./app/bitcoin/transactions");
var websocket_1 = require("./app/bitcoin/websocket");
var assets_1 = require("./app/liquid/assets");
var addresses_2 = require("./app/liquid/addresses");
var blocks_2 = require("./app/liquid/blocks");
var fees_2 = require("./app/liquid/fees");
var mempool_2 = require("./app/liquid/mempool");
var transactions_2 = require("./app/liquid/transactions");
var websocket_2 = require("./app/liquid/websocket");
var hostnameEndpointDefault = 'mempool.space';
var networkEndpointDefault = 'main';
var mempool = function (_a) {
    var _b = _a === void 0 ? {
        hostname: hostnameEndpointDefault,
        network: networkEndpointDefault,
    } : _a, hostname = _b.hostname, network = _b.network, protocol = _b.protocol, config = _b.config;
    if (!hostname)
        hostname = hostnameEndpointDefault;
    if (!network)
        network = networkEndpointDefault;
    var apiBitcoin = index_1.makeBitcoinAPI({
        hostname: hostname,
        network: network,
        protocol: protocol,
        config: config,
    }).api;
    var apiLiquid = index_1.makeLiquidAPI({
        hostname: hostname,
        network: network,
        protocol: protocol,
        config: config,
    }).api;
    return {
        bitcoin: {
            addresses: addresses_1.useAddresses(apiBitcoin),
            blocks: blocks_1.useBlocks(apiBitcoin),
            difficulty: difficulty_1.useDifficulty(apiBitcoin),
            fees: fees_1.useFees(apiBitcoin),
            lightning: lightning_1.useLightning(apiBitcoin),
            mempool: mempool_1.useMempool(apiBitcoin),
            transactions: transactions_1.useTransactions(apiBitcoin),
            websocket: websocket_1.useWebsocket(hostname, network, protocol),
        },
        liquid: {
            addresses: addresses_2.useAddresses(apiLiquid),
            assets: assets_1.useAssets(apiLiquid),
            blocks: blocks_2.useBlocks(apiLiquid),
            fees: fees_2.useFees(apiLiquid),
            mempool: mempool_2.useMempool(apiLiquid),
            transactions: transactions_2.useTransactions(apiLiquid),
            websocket: websocket_2.useWebsocket(hostname, network, protocol),
        },
    };
};
mempool.default = mempool;
module.exports = mempool;
